x/metadata/keeper/keeper.go

```diff
@@ -244,6 +244,26 @@ func (k Keeper) EmitEvent(ctx sdk.Context, event proto.Message) {
 	}
 }
 
+// unionUnique gets a union of the provided sets of strings without any duplicates.
+func (k Keeper) unionNoDups(sets ...[]string) []string {
```

Comment:
I know it might seem trivial, but shouldn't we write a simple test around this?

Reply:
Added some tests.